src/library/export/coverartcopyworker.cpp

```diff
@@ -0,0 +1,117 @@
+#include "library/export/coverartcopyworker.h"
+
+#include <QDebug>
+#include <QFileInfo>
+#include <QMessageBox>
+
+#include "util/imagefiledata.h"
+#include "util/safelywritablefile.h"
+
+void CoverArtCopyWorker::run() {
+    m_isCoverArtUpdated = false;
+    copyFile(m_coverArtImage, m_coverArtAbsolutePath);
+}
+
+void CoverArtCopyWorker::copyFile(
+        const ImageFileData& m_coverArtImage,
+        const QString& m_coverArtAbsolutePath) {
+    QFileInfo coverArtPathFileInfo(m_coverArtAbsolutePath);
+
+    if (coverArtPathFileInfo.exists()) {
+        switch (makeOverwriteRequest(m_coverArtAbsolutePath)) {
+        case OverwriteAnswer::CANCEL:
+            return;
+        case OverwriteAnswer::OVERWRITE:
+            break;
+        }
+
+        mixxx::SafelyWritableFile safelyWritableFile(m_coverArtAbsolutePath, false);
+        if (!safelyWritableFile.isReady()) {
+            qWarning()
+                    << "Unable to copy cover art into file"
+                    << m_coverArtAbsolutePath
+                    << "- Please check file permissions and storage space";
+            return;
+        }
```

Comment:
```suggestion
```

This can be removed we may assert the return value of `fileName().isEmpty()` later.